airbyte-cdk/python/airbyte_cdk/test/catalog_builder.py

```diff
@@ -0,0 +1,29 @@
+# Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+
+from typing import Any, Dict, List
+
+from airbyte_protocol.models import ConfiguredAirbyteCatalog, SyncMode
+
+
+class CatalogBuilder:
+    def __init__(self) -> None:
+        self._streams: List[Dict[str, Any]] = []
+
+    def with_stream(self, name: str, sync_mode: SyncMode) -> "CatalogBuilder":
+        self._streams.append(
+            {
+                "stream": {
+                    "name": name,
+                    "json_schema": {},
+                    "supported_sync_modes": ["full_refresh", "incremental"],
```

Comment:
fine for now but just flagging there might eventually be a need to make this field configurable since incremental is not supported by all streams

Reply:
Do we know of any source using this information from the catalog? It seems more like a platform information that the source doesn't care